storage/local/crashrecovery.go

```diff
@@ -211,35 +221,36 @@ func (p *persistence) sanitizeSeries(dirname string, fi os.FileInfo, fingerprint
 		if s == nil {
 			panic("fingerprint mapped to nil pointer")
 		}
-		if bytesToTrim == 0 && s.chunkDescsOffset != -1 &&
-			((s.headChunkPersisted && chunksInFile == s.chunkDescsOffset+len(s.chunkDescs)) ||
-				(!s.headChunkPersisted && chunksInFile == s.chunkDescsOffset+len(s.chunkDescs)-1)) {
+		if bytesToTrim == 0 && s.chunkDescsOffset != -1 && chunksInFile == s.chunkDescsOffset+s.persistWatermark {
 			// Everything is consistent. We are good.
 			return fp, true
 		}
 		// If we are here, something's fishy.
```

Comment:
Since "fishy" confused me, could you change this comment to "If we are here, we have to investigate further whether the series state is consistent." or something like that.